### PR TITLE
Timeperiod

### PR DESCRIPTION
**Pull request by kryofly**
Add an --timeperiod argument to hyperopt and backtesting.
The code was already there in testing, so just moved it in.

Example:
freqtrade backtesting --timeperiod -200
freqtrade hyperopt --timeperiod -200

Above examples will use the last 200 timeframes of tickerdata, instead of using the whole timespan.
This could be useful to quickly find out if an algorithm is feasible before expanding the testset.

### CODE REVIEW

File: freqtrade/misc.py [hunk 1]
```diff
@@ -211,6 +218,13 @@ def build_subcommands(parser: argparse.ArgumentParser) -> None:
         type=int,
         metavar='INT',
     )
+    hyperopt_cmd.add_argument(
+        '-tp', '--timeperiod',
```
Comment (gcarq) [inline freqtrade/misc.py]: Please use one character names for single hyphen parameters.

> In the original Unix tradition, command-line options are single letters preceded by a single hyphen.

See http://www.faqs.org/docs/artu/ch10s05.html

**Comment by kryofly** [comment]
I've removed the two-letter options, and skipping one-letter for now, to avoid a "land-grab" situation :)

**Comment by shusso** [comment]
@kryofly isn't there still [this](https://github.com/gcarq/freqtrade/pull/357/files#diff-e74a4dc95e212b8762dd770d4bebb573R222)

**Comment by glonlas** [comment]
LGTM.
What do you think @shusso

**Pull request merged**